docs/localization.md

```diff
@@ -0,0 +1,126 @@
+
+| Localization file name | Corresponding regional language variant | Subtags used |
+| ----------- | ----------- | ----------- |
+| `en.json` | English (default file)| en (English) |
+| `en-US.json` | American English | en (English) + US (United States) |
+| `en-AU.json` | Australian English | en (English) + AU (Australia) |
+| `fr.json` | French | en (French) |
+| `fr-CA.json` | Canadian French | fr (French) + CA (Canada) |
+
+### Translation keys
+
+The JSON files should contain key-value pairs for each locale. You can define translations based on a pre-defined keys used to translate the Catalyst storefront's [basic ecommerce functionality](https://www.catalyst.dev/docs#ecommerce-functionality). The translated values you specify will display to shoppers as static string translations.
+
+Use the existing `en.json` file as a template for the schema.
+You can only provide translated values for the translation keys specified in the template. 
+
+For example, the `en.json` file contains translation keys for 
+
+```json
+"Home": {
+  "Carousel": {
+    "bestSellingProducts": "Best Selling Products",
+    "featuredProducts": "Featured Products",
+    "newestProducts": "Newest products"
+  }
+}
+``` 
+
+In your newly-created JSON file, add a translation of the value to the new locale
+
+```json
+"Home": {
+  "Carousel": {
+    "bestSellingProducts": "Produits les plus vendus",
+    "featuredProducts": "Produits populaires",
+    "newestProducts": "Produits les plus recentsProduits"
+  }
+}
+``` 
+
+Read more about i18n messages in [next-intl docs](https://next-intl-docs.vercel.app/docs/usage/messages).
+
+## i18n configuration
+
+After you created a language file, add its name to the `i18n.ts` config file.
+
+For example, if you created a `fr.json` file, include the `fr` when you define the locales:   
+
+```ts
+const locales = ['en', 'fr'] as const;
+```
+
+## Using keys in React components
+
+The following example shows how messages can be used in React server and client components:
+
+```tsx
+import { NextIntlClientProvider } from 'next-intl';
+import { getMessages, getTranslations, unstable_setRequestLocale } from 'next-intl/server';
+// ...
+import { LocaleType } from '~/i18n';
+
+interface Props {
+  params: {
+    locale: LocaleType;
+  };
+}
+
+export default async function Home({ params: { locale } }: Props) {
+  unstable_setRequestLocale(locale);
+
+  const t = await getTranslations({ locale, namespace: 'Home' });
+  const messages = await getMessages({ locale });
+  // ...
+
+  return (
+    <div>
+      <NextIntlClientProvider locale={locale} messages={{ Product: messages.Product ?? {} }}>
+        <ProductCardCarousel
+          products={featuredProducts}
+          title={t('Carousel.featuredProducts')}
+        />
+      </NextIntlClientProvider>
```

Comment:
So this example is more aimed at using next-intl inside client components (which in turn use the `useTranslations` hook). I would demo how to us this in a server component (`getTranslations`) and ultimately how to use it in a client component (by using this provider like in this example).

Reply:
Yes, if someone could demo it to me and/or provide some example snippets that would be great

Reply:
@jorgemoya I've adjusted examples for server and client components. Let me know if it's good now.

Reply:
Looks good!